lib/deadpool-cyclone/src/instance/cyclone/local_uds.rs

```diff
@@ -486,6 +476,243 @@ fn temp_path_and_socket_from(
+                ];
+
+                let container_id = docker
+                    .create_container(
+                        Some(CreateContainerOptions {
+                            name: format!("cyclone-container-{rand_string}"),
+                            platform: Some(String::from("linux/amd64")),
+                        }),
+                        Config {
+                            image: Some(String::from("systeminit/cyclone")),
+                            cmd: Some(cmd),
+                            host_config: Some(HostConfig {
+                                mounts: Some(mounts),
+                                ..Default::default()
+                            }),
+                            ..Default::default()
+                        },
+                    )
+                    .await?
+                    .id;
+                Ok(LocalInstanceRuntime {
+                    runtime_strategy: LocalUdsRuntimeStrategy::LocalDocker,
+                    command: None,
+                    child: None,
+                    container_id: Some(container_id),
+                    docker_api: Some(docker),
+                })
+            }
+        }
+    }
+
+    async fn spawn(&mut self) -> result::Result<(), LocalUdsInstanceError> {
+        match &self.runtime_strategy {
+            LocalUdsRuntimeStrategy::LocalProcess => {
+                self.child = Some(
+                    self.command
+                        .as_mut()
+                        .expect("command should be available")
+                        .spawn()
+                        .map_err(LocalUdsInstanceError::ChildSpawn)?,
+                );
+                Ok(())
+            }
+            LocalUdsRuntimeStrategy::LocalDocker => {
+                self.docker_api
+                    .as_ref()
+                    .expect("docker socket should be available")
+                    .start_container(
+                        &self
+                            .container_id
+                            .clone()
+                            .expect("container id should be available"),
+                        None::<StartContainerOptions<String>>,
+                    )
+                    .await?;
+                Ok(())
+            }
+        }
+    }
+
+    async fn terminate(&mut self) -> result::Result<(), LocalUdsInstanceError> {
+        match &self.runtime_strategy {
+            LocalUdsRuntimeStrategy::LocalProcess => {
+                process::child_shutdown(
+                    self.child
+                        .as_mut()
+                        .expect("child process should be available"),
+                    Some(process::Signal::SIGTERM),
+                    None,
+                )
+                .await?;
+                Ok(())
+            }
+            LocalUdsRuntimeStrategy::LocalDocker => {
+                self.docker_api
+                    .as_mut()
+                    .expect("docker socket should be bound")
```

Comment:
I would not panic anywhere in the library if you can help it. It should be impossible, but eh??

Reply:
What's the right way to handle this? I'm fine propagating the error, I'm just not sure what the right pattern to do so is